core/src/main/java/com/ibm/wala/ipa/cha/ClassHierarchy.java

```diff
@@ -706,6 +708,56 @@ private void recursiveStringify(Node n, StringBuilder buffer) {
     }
   }
 
+  /**
+   * Converts ClassHierarchy to a JSON String, mapping each class name to a list of subclass names
+   */
+  public String toJson() {
+    // Initialize the map to store the <class, subclass> pairs
+    HashMap<String, Set<String>> classNameToSubclassNames = new HashMap<>();
+
+    // Start the recursive function from the root node
+    helperToJson(root, classNameToSubclassNames);
+
+    // Use Gson to convert the map to a JSON string
+    Gson gson = new Gson();
+    return gson.toJson(classNameToSubclassNames);
+  }
+
+  /** helper function to toJson that performs recursion to go through all of the DAG */
+  private void helperToJson(Node n, HashMap<String, Set<String>> hash) {
+    String key = nodeToString(n);
+
+    // If the node is already processed, return to avoid infinite recursion
+    if (hash.containsKey(key)) {
+      return;
+    }
+
+    // Initialize a set to store the names of the subclasses
+    Set<String> subclassNames = new HashSet<>();
+
+    // Process all children of the current node
+    if (n.children.size() > 0) {
```

Comment:
By printing out the JSON in the test and looking it over, I realized that it only handles subclassing, but not interface implementations.  E.g., for the interface `java.util.Map` we see `"java.util.Map":[]` in the output, missing all the classes that implement that interface.  We should update this logic so that for interfaces, we treat all implementing classes and sub-interfaces as subtypes in the outputted JSON.  In other words, the JSON should map each class or interface to all of its immediate subtypes, i.e., direct subclasses, direct sub-interfaces, or classes directly implementing an interface.